### PR TITLE
docs(misc) IgnorePlugin removed method. Chunk.files now a Set

### LINKED ISSUE #3119: Document webpack change: add more types and fix/remove TODOs
*A pull request by @sokra was merged and maintainers requested a documentation change.*

See pull request: https://github.com/webpack/webpack/pull/9277

---

**What kind of change does this PR introduce?**
refactoring

**Did you add tests for your changes?**
existing test

**Does this PR introduce a breaking change?**
yes

**What needs to be documented once your changes are merged?**
* IgnorePlugins `checkContext` was removed
* Chunk.files is now a Set

### PR DESCRIPTION
**Pull request by EugeneHlushko**
Closes #3119

### CODE REVIEW

**Comment by montogeek** [comment]
It is possible to say why it was removed and what is the alternative now?

**Comment by EugeneHlushko** [comment]
All i found was this: "remove checkContext, as checkResource already gets context"

File: src/content/plugins/ignore-plugin.md [hunk 1]
```diff
@@ -23,15 +23,11 @@ new webpack.IgnorePlugin(resourceRegExp, [contextRegExp]);
 
 ## Using filter functions
 
-- `checkContext(context)` A Filter function that receives context as the argument, must return boolean.
 - `checkResource(resource)` A Filter function that receives resource as the argument, must return boolean.
+- `checkContext(context)` was __removed in webpack 5__
```
Comment (EugeneHlushko) [inline src/content/plugins/ignore-plugin.md]:
```suggestion
- `checkContext(context)` was __removed in webpack 5__ as `checkResource` already gets context
```

File: src/content/plugins/ignore-plugin.md [hunk 2]
```diff
@@ -23,15 +23,11 @@ new webpack.IgnorePlugin(resourceRegExp, [contextRegExp]);
 
 ## Using filter functions
 
-- `checkContext(context)` A Filter function that receives context as the argument, must return boolean.
 - `checkResource(resource)` A Filter function that receives resource as the argument, must return boolean.
+- `checkContext(context)` was __removed in webpack 5__ as `checkResource` already gets context
```
Comment (EugeneHlushko) [inline src/content/plugins/ignore-plugin.md]:
```suggestion
- `checkContext(context)` was __removed in webpack 5__ as `checkResource` already gets context.
```

File: src/content/plugins/ignore-plugin.md [hunk 3]
```diff
@@ -23,15 +23,11 @@ new webpack.IgnorePlugin(resourceRegExp, [contextRegExp]);
 
 ## Using filter functions
 
-- `checkContext(context)` A Filter function that receives context as the argument, must return boolean.
 - `checkResource(resource)` A Filter function that receives resource as the argument, must return boolean.
```
Comment (EugeneHlushko) [inline src/content/plugins/ignore-plugin.md]:
```suggestion
- `checkResource (resource)` A Filter function that receives resource as the argument, must return boolean.
```

File: src/content/plugins/ignore-plugin.md [hunk 4]
```diff
@@ -23,15 +23,11 @@ new webpack.IgnorePlugin(resourceRegExp, [contextRegExp]);
...
+- `checkContext(context)` was __removed in webpack 5__ as `checkResource` already gets context.
```
Comment (EugeneHlushko) [inline src/content/plugins/ignore-plugin.md]:
```suggestion
- `checkContext (context)` was __removed in webpack 5__ as `checkResource` already gets context.
```

File: src/content/plugins/ignore-plugin.md [hunk 5]
```diff
@@ -23,15 +23,11 @@ new webpack.IgnorePlugin(resourceRegExp, [contextRegExp]);
 
 ## Using filter functions
 
-- `checkContext(context)` A Filter function that receives context as the argument, must return boolean.
-- `checkResource(resource)` A Filter function that receives resource as the argument, must return boolean.
+- `checkResource (resource)` A Filter function that receives resource as the argument, must return boolean.
```
Comment (EugeneHlushko) [inline src/content/plugins/ignore-plugin.md]:
```suggestion
- `checkResource (resource, context)` A Filter function that receives `resource` and `context` as arguments, must return boolean.
```

**Comment by EugeneHlushko** [comment]
@montogeek now it looks better, ty!

**Comment by montogeek** [comment]
Thanks!

**Pull request merged**